packages/intent-filler/src/strategies/basic.ts

```diff
@@ -0,0 +1,441 @@
+import { FillerStrategy } from "@/strategies/base"
+import { Order, FillerConfig, ExecutionResult, HexString, FillOptions, PaymentInfo } from "@/types"
+import { ethers } from "ethers"
+import { encodePacked, keccak256, toHex } from "viem"
+import { ADDRESS_ZERO, fetchTokenUsdPriceOnchain, getOrderCommitment } from "@/utils"
+import { INTENT_GATEWAY_ABI } from "@/config/abis/IntentGateway"
+import { ERC20_ABI } from "@/config/abis/ERC20"
+import { addresses, assets } from "@/config/chain"
+
+export class BasicFiller implements FillerStrategy {
+	name = "BasicFiller"
+
+	private wallet: ethers.Wallet
+	private provider: ethers.providers.Provider
+	private contract: ethers.Contract
+	private chain: string
+
+	constructor(chain: string, privateKey: string, provider: ethers.providers.Provider, contractAddress: string) {
+		this.chain = toHex(chain)
+		this.provider = provider
+		this.wallet = new ethers.Wallet(privateKey, provider)
+		this.contract = new ethers.Contract(contractAddress, INTENT_GATEWAY_ABI, this.wallet)
+	}
+
+	/**
+	 * Determines if this strategy can fill the given order
+	 * @param order The order to check
+	 * @param config The filler configuration
+	 * @returns True if the strategy can fill the order
+	 */
+	async canFill(order: Order, config: FillerConfig): Promise<boolean> {
+		try {
+			const currentChain = this.chain
+			const destChain = toHex(order.destChain)
+
+			if (!destChain.includes(currentChain)) {
+				console.debug(`Order destined for chain ${destChain}, we're on ${currentChain}`)
+				return false
+			}
+
+			const currentBlock = await this.provider.getBlockNumber()
+			const deadline = BigInt(order.deadline)
+
+			if (deadline < currentBlock) {
+				console.debug(`Order expired at block ${deadline}, current block ${currentBlock}`)
+				return false
+			}
+
+			const isAlreadyFilled = await this.checkIfOrderFilled(order)
+			if (isAlreadyFilled) {
+				console.debug(`Order is already filled`)
+				return false
+			}
+
+			const hasEnoughTokens = await this.checkTokenBalances(order.outputs)
+			if (!hasEnoughTokens) {
+				console.debug(`Insufficient token balances for order`)
+				return false
+			}
+
+			const orderValue = this.calculateOrderValue(order)
+			const requiredConfirmations = config.confirmationPolicy.getConfirmationBlocks(
```

Comment:
The confirmation Policy is a feature of the source chain, not the destination chain, It is basically the number of confirmation blocks on the source chain after the order has been placed before it's considered for filling on the destination